cli/sub_command.go

```diff
@@ -306,7 +342,17 @@ func (c *subCmd) subscribe(p *fisk.ParseContext) error {
 		return err
 	}
 
-	<-ctx.Done()
+	var tc <-chan time.Time
+	if t != nil {
+		tc = t.C
+	} else {
+		tc = make(chan time.Time)
+	}
+
+	select {
+	case <-ctx.Done():
+	case <-tc:
+	}
```

Comment:
Just to clarify the intent of "wait," it is the wait time _after_ the last message received (maybe timeout is a better option name). So that is why it gets reset any time a message is received.

For the `done` chan, I suppose I could also simply call `cancel()` from within the `AfterFunc` function? It could be reworked that once `<-ctx.Done()` is received on, `sub.Unsubscribe` is called as well.

Reply:
Right missed that subtlety, this makes more sense now sorry about that.

Yes could just call cancel() indeed, unsubscribe is kind of redundant since we're exiting - would be a wasted round trip.